packages/aws-lambda-otel-extension/opt/otel-extension/external/helper.js

```diff
@@ -10,6 +11,44 @@ const extensionVersion = (() => {
   return require('../version');
 })();
 
+const isJSONResponse = (responseData) => responseData && typeof responseData === 'object';
+
+const isHttpResponse = (responseData) =>
+  isJSONResponse(responseData) && typeof get(responseData, 'body') === 'string';
+
+const cleanResponseData = (data) => {
+  let response = '';
+  data.excludedBody = false;
+
+  let responseData = data.responseData;
+  if (typeof responseData === 'string') {
+    try {
+      responseData = JSON.parse(responseData);
+    } catch (error) {
+      /* noop */
+    }
+  }
+
+  // Only save responses if it is an object
+  if (isHttpResponse(responseData)) {
+    try {
+      JSON.parse(responseData.body);
+      response = JSON.stringify(responseData);
+    } catch (error) {
+      responseData.body = '';
+      response = JSON.stringify(responseData);
+      data.excludedBody = true;
+    }
+  } else if (isJSONResponse(responseData)) {
+    response = JSON.stringify(responseData);
+  } else {
+    data.excludedBody = true;
+  }
+
+  data.responseData = response;
+  return data;
```

Comment:
How about going with approach, where we do not mix _raw_ and parsed out of JSON response under same variable, so it's clear in all places what value is behind the variables, e.g.:


```javascript
const raw = data.responseData;

const json = resolveJsonOrNull(raw);
if (!json) {
  if (typeof raw === 'string') delete data.responsData;
  return data;
}
const jsonHttpResponse = resolveJsonOrNull(json.body);
if (!jsonHttpResponse) return data;
delete json.body;
return Object.assign(data, {
  excludedBody: true,
  responseData: JSON.stringify(json)
});
```